PayPal/src/main/java/com/braintreepayments/api/paypal/PayPalInternalClient.kt

```diff
@@ -131,6 +141,110 @@ internal class PayPalInternalClient(
             ?: redirectUri.getQueryParameter("token")
     }
 
+    @ExperimentalBetaApi
+    fun sendVaultEditRequest(
+        context: Context,
+        payPalVaultEditRequest: PayPalVaultEditRequest,
+        callback: PayPalVaultEditCallback
+    ) {
+        getClientMetadataId(
+            context
+        ) { clientMetadataId ->
+            if (clientMetadataId == null) {
+                val result = PayPalVaultEditResponse.Failure(
+                    BraintreeException("Could not retrieve clientMetaDataId")
+                )
+                callback.onPayPalVaultEditResult(result)
+            } else {
+
+                val riskCorrelationId = (payPalVaultEditRequest as? PayPalVaultErrorHandlingEditRequest)?.riskCorrelationId ?: clientMetadataId
+
+                sendVaultEditRequestWithRiskCorrelationId(
+                    context,
+                    payPalVaultEditRequest,
+                    riskCorrelationId,
+                    callback
+                )
+            }
+        }
+    }
+
+    @ExperimentalBetaApi
+    private fun sendVaultEditRequestWithRiskCorrelationId(
+        context: Context,
+        payPalVaultEditRequest: PayPalVaultEditRequest,
+        riskCorrelationId: String,
+        callback: PayPalVaultEditCallback
+    ) {
+        val params = parameters(payPalVaultEditRequest.editPayPalVaultId).toMutableMap()
+
+        params["risk_correlation_id"] = riskCorrelationId
+
+        val jsonObject = JSONObject(params.toMap())
+
+        braintreeClient.sendPOST(
+            payPalVaultEditRequest.hermesPath,
+            jsonObject.toString()
+        ) { response, error ->
+            if (error != null) {
+                val result = PayPalVaultEditResponse.Failure(error, riskCorrelationId)
+                callback.onPayPalVaultEditResult(result)
+            } else {
+                try {
+                    val responseBody = JSONObject(response)
+                    val agreementSetup = responseBody.getJSONObject("agreementSetup")
+
+                    val editFIAgreementSetup = EditFIAgreementSetup(
+                        agreementSetup.getString("tokenId"),
+                        agreementSetup.getString("approvalUrl"),
+                        agreementSetup.getString("paypalAppApprovalUrl")
```

Comment:
We can remove this, this is for app switch and we should not check for it in this flow currently